lib/cf-socket.c

```diff
@@ -146,7 +146,10 @@ static void nosigpipe(struct Curl_easy *data,
 #define nosigpipe(x,y) Curl_nop_stmt
 #endif
 
-#if defined(USE_WINSOCK) || \
+#if defined(USE_WINSOCK) && defined(TCP_KEEPIDLE) && defined(TCP_KEEPINTVL) && defined(TCP_KEEPCNT)
+#define WINSOCK_SUPPORTS_KEEP_IDLE_INTVL_CNT
+#endif
+#if (defined(USE_WINSOCK) && !defined(WINSOCK_SUPPORTS_KEEP_IDLE_INTVL_CNT)) || \
```

Comment:
How about instead doing it with an `#elif`? I think it makes it a little clearer. Like:

~~~c
#if defined(USE_WINSOCK) && defined(TCP_KEEPIDLE) && defined(TCP_KEEPINTVL) && defined(TCP_KEEPCNT)
#define WINSOCK_SUPPORTS_KEEP_IDLE_INTVL_CNT
#elif defined(USE_WINSOCK) || \
...

Reply:
Yes elif looks better.